reasoner/ReasonerProducer.java

```diff
@@ -69,25 +70,25 @@ private void requestAnswered(ResolutionAnswer answer) {
     }
 
     private void requestFailed(int iteration) {
```

Comment:
Actually, I would prefer that this method be called `requestExhausted`. Otherwise we suddenly switch terminology and I don't see any reason to!